### PR TITLE
Add AzureRM 2.0.x documentation

### PR DESCRIPTION
**Pull request by apuntamb**

### CODE REVIEW

File: plugins/foreman_azure/2.x/index.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+version: 2.x
+# uncomment for older versions than stable or nightly
+#robots: noindex
+---
+
+# 1. Basics
+
+Foreman AzureRM is the plugin for Azure v2 i.e. Azure Resource Manager API. It allows you to provision and manage your hosts in Azure Resource Manager cloud environment. Users do have restricted access within the Azure subscription (such that only they can create hosts) based on Resource Groups in AzureRM.
+
+This plugin 2.x series has no dependency of foreman_azure 1.x as they refer to different Azure APIs. This 2.x should be compatible with Foreman 1.17 onwards but, it recommended to use latest foreman version.
```
Comment (upadhyeammit) [inline plugins/foreman_azure/2.x/index.md]: Replace foreman with Foreman

File: plugins/foreman_azure/2.x/index.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+Then run `bundle install` from the same directory
+
+# 3. Configuration
+
+Please refer to our documentation [Github](https://github.com/theforeman/foreman_azure_rm#foreman-azurerm-plugin)
+
+# 4. Usage
+
+Once the compute resource with AzureRM provider is created, next step is Host creation. You have to select your AzureRM compute resource on the Hosts page and the relevant fields shall appear on the Virtual Machine tab. You can now fill in your resource group, size of the VM you want to provision, username and password or sshkey with which the user can login to the provisioned machine. In addition to that, you can provide any custom script or command to be executed during the VM creation process. In the Operating Systems tab, you should specify the Architecture, OS, Root Password and the relevant Image that you have created under the compute resource. Last specification is for the Network Interface. You should select a domain defined under Foreman and then select Azure subnet and Public or Private IP format. Upon clicking submit, the host will be successfully provisioned and VM details will then be available.
```
Comment (upadhyeammit) [inline plugins/foreman_azure/2.x/index.md]: Is it mandatory to provide root password ?
I feel Usage would look clean if we use bullets or numbering ?
With this I feel we should clearly mention what is mandatory and what is not ? I know WebUI covers this but at same time having that in documentation is good idea ?

**Review by upadhyeammit — COMMENTED** [review]
Added few comments, rest looks good .

File: plugins/foreman_azure/2.x/index.md (hunk 2 above)
Comment (apuntamb) [reply plugins/foreman_azure/2.x/index.md]: Yes sure, this makes sense.

**Comment by tbrisker** [comment]
@ShimShtein can you please take a look as well?
@apuntamb - you should also add a link to this from the headline feature in 1.24

File: plugins/foreman_azure/2.x/index.md (hunk 2 above)
Comment (ShimShtein) [inline plugins/foreman_azure/2.x/index.md]: Probably worth adding a couple of sentences about how to define a new compute resource, especially the fact that there should be created a resource per region.

**Review by tbrisker — CHANGES_REQUESTED** [review]
@apuntamb - You should also add a link to this page from the main azure plugin page and uncomment `robots: noindex` on the 1.x page so it stops being indexed on google. Also see @ShimShtein's comment inline that wasn't addressed.

**Comment by apuntamb** [comment]
@upadhyeammit tailored the writing as per your suggestions.
@ShimShtein added how is an azurerm compute resource now defined.
@tbrisker added the link to headline feature in index.md and uncommented `robots:noindex`. Just a query though, I am afraid I didn't understand: `You should also add a link to this page from the main azure plugin page`.

Note: I have also changed the directory name to 2.0.x since this documentation will be common for all the minor releases of foreman_azure_rm starting from 2.0.1 to upcoming releases till 2.0.x. Please let me know if we could make this better.

File: plugins/foreman_azure/2.0.x/index.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+version: 2.0.x
+# uncomment for older versions than stable or nightly
+#robots: noindex
+---
+
+# 1. Basics
+
+Foreman AzureRM is the plugin for Azure v2 i.e. Azure Resource Manager API. It allows you to provision and manage your hosts in Azure Resource Manager cloud environment. Users do have restricted access within the Azure subscription (such that only they can create hosts) based on Resource Groups in AzureRM.
+
+This plugin 2.0.x series has no dependency of foreman_azure 1.x as they refer to different Azure APIs. This 2.0.x should be compatible with Foreman 1.17 onwards but, it recommended to use latest Foreman version. Foreman AzureRM is also one of the [headline features](https://www.theforeman.org/manuals/1.24/index.html#Headlinefeatures) for Foreman 1.24.
```
Comment (ShimShtein) [inline plugins/foreman_azure/2.0.x/index.md]:
```suggestion
This plugin 2.0.x series has no dependency of foreman_azure 1.x as they refer to different Azure APIs. This 2.0.x should be compatible with Foreman 1.17 onwards, but it is recommended to use latest Foreman version. Foreman AzureRM is also one of the [headline features](https://www.theforeman.org/manuals/1.24/index.html#Headlinefeatures) for Foreman 1.24. 
```

**Review by ShimShtein — CHANGES_REQUESTED** [review]

File: plugins/foreman_azure/2.0.x/index.md (hunk 3 above)
Comment (apuntamb) [reply plugins/foreman_azure/2.0.x/index.md]: done.

**Comment by apuntamb** [comment]
@ShimShtein @tbrisker Also, should we mark or add "deprecated" for the 1.x version? Since with the same installer command, it shall now install only azure_rm(new) and 1.x documentation mentions that old installer command to install azure(old) which might cause a bit of ambiguity for users. 

Let me know what you guys think.

File: plugins/foreman_azure/2.0.x/index.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+Set up the repo as explained in the link above, then run
+
+    # yum install tfm-rubygem-foreman_azure_rm
+
+### Fedora (rpm)
+
+Set up the repo as explained in the link above, then run
+
+    # yum install rubygem-foreman_azure_rm
```
Comment (tbrisker) [inline plugins/foreman_azure/2.0.x/index.md]: We don't build plugins (or foreman for that matter) for fedora.
What about debian?

File: plugins/foreman_azure/2.0.x/index.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    $ gem 'foreman_azure_rm'
+
+Then run `bundle install` from the same directory
+
+# 3. Configuration
+
+Please refer to our documentation [Github](https://github.com/theforeman/foreman_azure_rm#foreman-azurerm-plugin)
+
+# 4. Usage
+* The Compute Resource is created using the four specified credentials viz., Client ID, Client Secret, Subscription ID and Tenant ID.
```
Comment (tbrisker) [inline plugins/foreman_azure/2.0.x/index.md]:
```suggestion
* Creating the Compute Resource requires using four credentials: Client ID, Client Secret, Subscription ID and Tenant ID.
```

File: plugins/foreman_azure/2.0.x/index.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+* In addition to this, the compute resource is created per region. This means, you must specify a region from the regions available and the cloud instances will then be created based on the region selected for the compute resource.
```
Comment (tbrisker) [inline plugins/foreman_azure/2.0.x/index.md]:
```suggestion
* A separate compute resource should be created per region. This means a region must be specified from the regions available and the cloud instances will then be created based on the region selected for the compute resource.
```

File: plugins/foreman_azure/2.0.x/index.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+See the [troubleshooting section]({{site.baseurl}}manuals/latest/index.html#7.2GettingHelp)
+in the Foreman manual for more info.
+
+## 5.2 Contributing
+
+Follow the [same process as Foreman]({{site.baseurl}}contribute.html#SubmitPatches)
+for contributing.
+
+Sources at [github.com/theforeman/foreman_azure_rm](https://github.com/theforeman/foreman_azure_rm)
```
Comment (tbrisker) [inline plugins/foreman_azure/2.0.x/index.md]:
```suggestion
The source code for the plugin can be found at [github.com/theforeman/foreman_azure_rm](https://github.com/theforeman/foreman_azure_rm).
```

**Review by tbrisker — CHANGES_REQUESTED** [review]
A few inline comments. Additionally, can you please link to the manual from https://github.com/theforeman/theforeman.org/blob/gh-pages/_includes/manuals/1.24/1.2_release_notes.md#support-for-azure-resource-manager-compute-resource ?

File: plugins/foreman_azure/2.0.x/index.md (hunk 4 above)
Comment (apuntamb) [reply plugins/foreman_azure/2.0.x/index.md]: About fedora, I assumed so as I could see stated the same in quite some documentations, for eg. [foreman_discovery](https://github.com/theforeman/theforeman.org/blob/gh-pages/plugins/foreman_discovery/15.0/index.md#212-manual-installation), it says for RHEL & Fedora.
About Debian, we haven't packaged it for debian yet.

File: _includes/manuals/1.24/1.2_release_notes.md [hunk 8]
```diff
@@ -2,7 +2,7 @@
 
 ### Headline features
 #### Support for Azure Resource Manager Compute Resource
-Foreman can now provision hosts using Azure Resource Manager (AzureRM) as a Compute Resource via the `foreman_azure_rm` plugin. Provisioning can be successfully done on Linux platforms with either Finish template or Userdata template. Currently, it supports single OS disk for the VM and uses public images provided by Azure Marketplace.
+Foreman can now provision hosts using Azure Resource Manager (AzureRM) as a Compute Resource via the `foreman_azure_rm` plugin. Provisioning can be successfully done on Linux platforms with either Finish template or Userdata template. Currently, it supports single OS disk for the VM and uses public images provided by Azure Marketplace. For more information, head over to [Foreman Azure](https://theforeman.org/plugins/foreman_azure/2.0.x/index.html) manual.
```
Comment (apuntamb) [inline _includes/manuals/1.24/1.2_release_notes.md:5]: @tbrisker Did you mean adding the manual's link here in the headline's section, like this?

Comment (tbrisker) [reply _includes/manuals/1.24/1.2_release_notes.md:5]: exactly! :1st_place_medal:

File: plugins/foreman_azure/2.0.x/index.md (hunk 4 above)
Comment (tbrisker) [reply plugins/foreman_azure/2.0.x/index.md]: I guess those are leftovers from the days we did packaging for fedora as well (up until 1.15). could you do a clean up in a separate pr for the other plugins? doesn't have to be for all versions, just the latest one of each.

File: plugins/foreman_azure/2.0.x/index.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+* You can now fill in your resource group, size of the VM you want to provision, username and password or sshkey with which the user can login to the provisioned machine.
+
+* In addition to that, you can provide any custom script or command to be executed during the VM creation process.
+
+* In the Operating Systems tab, you must specify the Architecture, OS and the relevant Image that you have created under the compute resource. Although, setting up Root Password is not mandatory.
+
+* Last specification is for the Network Interface. You should select a domain defined under Foreman and then select Azure subnet and Public or Private IP format.
+
+* Upon clicking submit, the host will be successfully provisioned and VM details will then be available.
```
Comment (upadhyeammit) [inline plugins/foreman_azure/2.0.x/index.md]: Upon clicking Submit provisioning should start and VM details will be available on successful provisioning.

File: plugins/foreman_azure/2.0.x/index.md (hunk 4 above)
Comment (apuntamb) [reply plugins/foreman_azure/2.0.x/index.md]: Ahh Okk. Yeah sure makes sense.
+1

**Comment by apuntamb** [comment]
@tbrisker @upadhyeammit @ShimShtein Addressed all the above mentioned changes.

**Review by upadhyeammit — APPROVED** [review]
LGTM!

**Review by tbrisker — APPROVED** [review]
thanks @apuntamb !

**Pull request merged**